claat/cmd/fetch.go

```diff
@@ -64,17 +64,20 @@ type codelab struct {
 }
 
 // slurpCodelab retrieves and parses codelab source.
+// It takes the source, plus an auth token and a set of extra metadata to pass along.
 // It returns parsed codelab and its source type.
 //
 // The function will also fetch and parse fragments included
 // with types.ImportNode.
-func slurpCodelab(src, authToken string) (*codelab, error) {
+func slurpCodelab(src, authToken string, passMetadata map[string]bool) (*codelab, error) {
 	res, err := fetch(src, authToken)
 	if err != nil {
 		return nil, err
 	}
 	defer res.body.Close()
-	clab, err := parser.Parse(string(res.typ), res.body)
+	clab, err := parser.Parse(string(res.typ), res.body, parser.Options{
+		PassMetadata: passMetadata,
```

Comment:
This multiline declaration looks like JavaScript. I'd make the `parser.Options` call into a constructor.

maybe-out-of-scope-or-too-nit: smelly params of inline conversion to a string + passing another struct variable as param ... I'd make into `parseParams` helper f(x) or something.

Reply:
There's nothing inherently wrong with a multiline declaration. A NewOptions function is called for because the map needs to be zeroed.